src/daemon/devnode.c

```diff
@@ -36,6 +36,33 @@ int rm_recursive(const char* path){
     return remove(path);
 }
 
+int check_chown(const char *pathname, uid_t owner, gid_t group) {
+    if (chown(pathname, owner, group)) {
```

Comment:
As far as I can tell, this condition is wrong; `chown` returns 0 on success, but -1 on failure, so this path is taken when `chown` fails, and the failure path is taken when it succeeds.

Additionally, I would rather have this function worded like this:
```c
if (function() < 0) {
    ckb_warn("Oh noes!\n");
    return -1;
}
return 0;
```